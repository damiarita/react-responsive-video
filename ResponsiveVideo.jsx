import React from 'react';

const ResponsiveVideo = () => {
      return (
        <picture/>
      );
    }
  
  export default ResponsiveVideo;