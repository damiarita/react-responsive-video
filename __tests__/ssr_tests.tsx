/**
 * @jest-environment node
 */
import ResponsiveVideo from '../src';
import { pictureProps, imgProps, videoProps, sizes } from './data/fullProps';
import { sizes as minimalSizes } from './data/minProps';
import { sizes as invalidSizes } from './data/invalidProps';
import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';

describe('The behaviour of the component on SSR', () => {
  it('SSR renders correctly when all props are sent', () => {
    const component = renderToStaticMarkup(
      <ResponsiveVideo
        pictureProps={pictureProps}
        imgProps={imgProps}
        videoProps={videoProps}
        sizes={sizes}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('SSR renders correctly when minimum props are sent', () => {
    const component = renderToStaticMarkup(
      <ResponsiveVideo sizes={minimalSizes} />,
      { ssr: true },
    );
    expect(component).toMatchSnapshot();
  });

  it('SSR renders correctly when invalid props are sent', () => {
    const component = renderToStaticMarkup(
      <ResponsiveVideo sizes={invalidSizes} />,
    );
    expect(component).toMatchSnapshot();
  });
});
