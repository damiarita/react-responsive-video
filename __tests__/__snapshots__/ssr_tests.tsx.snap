// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR renders correctly 1`] = `
<picture>
  <source
    height={972}
    media="(min-width: 1200px)"
    srcSet="https://www.example.com/desktop.avif"
    type="image/avif"
    width={2000}
  />
  <source
    height={972}
    media="(min-width: 1200px)"
    srcSet="https://www.example.com/desktop.webp"
    type="image/webp"
    width={2000}
  />
  <source
    height={972}
    media="(min-width: 1200px)"
    srcSet="https://www.example.com/desktop.jpg"
    type="image/jpeg"
    width={2000}
  />
  <source
    height={1024}
    media="(min-width: 768px)"
    srcSet="https://www.example.com/tablet.avif"
    type="image/avif"
    width={1024}
  />
  <source
    height={1024}
    media="(min-width: 768px)"
    srcSet="https://www.example.com/tablet.webp"
    type="image/webp"
    width={1024}
  />
  <source
    height={1024}
    media="(min-width: 768px)"
    srcSet="https://www.example.com/tablet.jpg"
    type="image/jpeg"
    width={1024}
  />
  <source
    height={900}
    srcSet="https://www.example.com/mobile.avif"
    type="image/avif"
    width={600}
  />
  <source
    height={900}
    srcSet="https://www.example.com/mobile.webp"
    type="image/webp"
    width={600}
  />
  <source
    height={900}
    srcSet="https://www.example.com/mobile.jpg"
    type="image/jpeg"
    width={600}
  />
</picture>
`;
